### PR TITLE
To address #157,#158,#159, #150, #149 corresponding to WGLC comments 

### PR DESCRIPTION
**Pull request by inband-oam**
Addressing #157,#158,#159, #150, #149

### CODE REVIEW

File: drafts/draft-ietf-ippm-ioam-data.xml [hunk 1]
```diff
@@ -2024,16 +2039,13 @@ be 4-octet aligned:
       comments and advice.</t>
 
       <t>This document leverages and builds on top of several concepts
-      described in <xref target="I-D.kitamura-ipv6-record-route"/>. The
+      described in <xref target="I-D.kitamura-ipv6-record-route"></xref>. The
       authors would like to acknowledge the work done by the author Hiroshi
       Kitamura and people involved in writing it.</t>
 
       <t>The authors would like to gracefully acknowledge useful review and
       insightful comments received from Joe Clarke, Al Morton, Tom Herbet,
```
Comment (mickeyspiegel) [inline drafts/draft-ietf-ippm-ioam-data.xml]: s/Tom Herbet/Tom Herbert

File: drafts/draft-ietf-ippm-ioam-data.xml [hunk 2]
```diff
@@ -2024,16 +2039,13 @@ be 4-octet aligned:
...
-      Haoyu song, and Mickey Spiegel.</t>
-
-      <t>The authors would like to acknowledge the contribution of "Immediate
-      export" of IOAM trace by Barak Gafni.</t>
+      Haoyu song, Mickey Spiegel and Barak Gafni.</t>
```
Comment (mickeyspiegel) [inline drafts/draft-ietf-ippm-ioam-data.xml]: nit: s/song/Song

File: drafts/draft-ietf-ippm-ioam-ipv6-options.xml [hunk 3]
```diff
@@ -0,0 +1,499 @@
...
+          <street>4750 Patrick Henry Drive</street>
+
+          <city>Santa Clara, CA</city>
+
+          <code>95054</code>
+
+          <country>US</country>
+        </postal>
+
+        <email>mspiegel@barefootnetworks.com</email>
```
Comment (mickeyspiegel) [inline drafts/draft-ietf-ippm-ioam-ipv6-options.xml]: mickey.spiegel@intel.com

File: drafts/draft-ietf-ippm-ioam-ipv6-options.xml [hunk 4]
```diff
@@ -0,0 +1,499 @@
...
+
+          <country>US</country>
+        </postal>
+
+        <email>petr@fb.com</email>
+      </address>
+    </author>
+
+    <author fullname="Mickey Spiegel" initials="M." surname="Spiegel">
+      <organization abbrev="">Barefoot Networks</organization>
```
Comment (mickeyspiegel) [inline drafts/draft-ietf-ippm-ioam-ipv6-options.xml]: Barefoot Networks, an Intel company

File: drafts/draft-ietf-ippm-ioam-ipv6-options.xml [hunk 5]
```diff
@@ -0,0 +1,499 @@
...
+      particular extension header type depends on IOAM usage, as described in
+      section 4 of <xref target="I-D.ietf-ippm-ioam-data"></xref>. Multiple
+      options with the same Option Type MAY appear in the same Hop-by-Hop
+      Options or Destination Options header, with varying content.</t>
+
+      <t>In order for IOAM to work in IPv6 networks, IOAM MUST be explicitly
+      enabled per interface on every node within the IOAM domain. Unless a
+      particular interface is explicitly enabled (i.e. explicitly configured)
+      for IOAM, a router MUST drop packets which contain extension headers
+      carrying IOAM data-fields. This is the default behavior and is
```
Comment (mickeyspiegel) [inline drafts/draft-ietf-ippm-ioam-ipv6-options.xml]: We can specify it this way, but if we do shouldn't the requested Option Type "codepoint" other than "00"?
See comment below.

File: drafts/draft-ietf-ippm-ioam-ipv6-options.xml [hunk 6]
```diff
@@ -0,0 +1,499 @@
...
+          Option-Type-specific data.</t>
+        </list></t>
+
+      <t>In-situ OAM Options are inserted as Option data as follows:<list
+          style="numbers">
+          <t>Pre-allocated Tracing Option: The in-situ OAM Preallocated
+          Tracing option defined in <xref
+          target="I-D.ietf-ippm-ioam-data"></xref> is represented as a IPv6
+          option in hop by hop extension header: <list style="hanging">
+              <t hangText="Option Type:">001xxxxx 8-bit identifier of the IOAM
```
Comment (mickeyspiegel) [inline drafts/draft-ietf-ippm-ioam-ipv6-options.xml]: Why start with "00" if the default behavior is to drop packets which contain extension headers carrying IOAM data-fields?
"00" means when the option type is unrecognized, "skip over this option and continue processing the header".
The other possibilities, any of which would be consistent with the default behavior described above, are:

- "01" (discard the packet),
- "10" (discard and send ICMP parameter problem, code 2, back to the packet's source address), or
- "11" (discard and send ICMP only if the packet's destination address was not a multicast address).

File: drafts/draft-ietf-ippm-ioam-data.xml [hunk 7]
```diff
@@ -934,21 +943,24 @@ The trace option data MUST be 4-octet aligned:
...
+
+              <t hangText="Node data List [n]:">Variable-length field. This is
+              a list of node data elements where the content of each node data
+              element is determined by the IOAM-Trace-Type. The order of
+              packing the data fields in each node data element follows the
+              bit order of the IOAM-Trace-Type field. The last node data
+              element in this list is the node data of the first IOAM capable
+              node in the path. If the nth element of this list is populated
+              by a node, the next IOAM capable node in the path populates
+              (n-1)th element in this list. Populating the node data list in
```
[the remark this answers is not in the document]
Comment (inband-oam) [reply drafts/draft-ietf-ippm-ioam-data.xml]: "such that the transmitted node data list
begins with this node's data element as the first element in the list. "
is still confusing - in the pre-allocated while not the first element in the list, it is the first populated element in the list. 
would:
Each node MUST prepend its node data element in front of the node data
elements that it received, such that the transmitted node data list
begins with this node's data element as the first _populated_ element in the list.

File: drafts/draft-ietf-ippm-ioam-data.xml [hunk 8]
```diff
@@ -343,43 +343,49 @@
...
+      multiple paths, or scenarios in which probe traffic is potentially
+      handled differently from regular data traffic by the network
+      devices.</t>
+
+      <t>IOAM usecases and mechanisms have expanded as this document matured
+      and have resulted in additional flags and options that may result in
+      creation of additional packets dedicated to OAM. But the term IOAM
+      continues to be used where "in-situ" refers to the OAM data added to the
+      data packets that controls the content and additional OAM packet
+      creation.</t>
```
[the remark this answers is not in the document]
Comment (inband-oam) [reply drafts/draft-ietf-ippm-ioam-data.xml]: addressed, the text suggestion looks good to me.

File: drafts/draft-ietf-ippm-ioam-data.xml [hunk 9]
```diff
@@ -1471,9 +1483,9 @@ IOAM proof of transit option of IOAM POT Type 0:
               configured parameters.</t>
             </list>Note: Larger or smaller sizes of "Random" and "Cumulative"
           data are feasible and could be required for certain deployments
-          (e.g. in case of space constraints in the transport protocol used).
-          Future versions of this document will address different sizes of
-          data for "proof of transit".</t>
+          (e.g. in case of space constraints in the encapsulation protocols
+          used). Future document may address different sizes of data for
```
[the remark this answers is not in the document]
Comment (inband-oam) [reply drafts/draft-ietf-ippm-ioam-data.xml]: done!

File: drafts/draft-ietf-ippm-ioam-data.xml (hunk 1 above)
Comment (inband-oam) [reply drafts/draft-ietf-ippm-ioam-data.xml]: done, thanks for noticing it!

File: drafts/draft-ietf-ippm-ioam-ipv6-options.xml (hunk 3 above)
Comment (inband-oam) [reply drafts/draft-ietf-ippm-ioam-ipv6-options.xml]: Sorry, draft-ietf-ippm-ioam-ipv6-options.xml  changes showed up in the pull request as my local master  was not in sync with the repo. Now merged to latest. Only intended to address data draft changes in this pull request.

File: drafts/draft-ietf-ippm-ioam-ipv6-options.xml (hunk 5 above)
Comment (inband-oam) [reply drafts/draft-ietf-ippm-ioam-ipv6-options.xml]: Sorry, draft-ietf-ippm-ioam-ipv6-options.xml changes showed up in the pull request as my local master was not in sync with the repo. Now merged to latest. Only intended to address data draft changes in this pull request.

File: drafts/draft-ietf-ippm-ioam-ipv6-options.xml (hunk 6 above)
Comment (inband-oam) [reply drafts/draft-ietf-ippm-ioam-ipv6-options.xml]: Sorry, draft-ietf-ippm-ioam-ipv6-options.xml changes showed up in the pull request as my local master was not in sync with the repo. Now merged to latest. Only intended to address data draft changes in this pull request.

File: drafts/draft-ietf-ippm-ioam-data.xml (hunk 7 above)
[the remark this answers is not in the document]
Comment (mickeyspiegel) [reply drafts/draft-ietf-ippm-ioam-data.xml]: LGTM

File: drafts/draft-ietf-ippm-ioam-data.xml [hunk 10]
```diff
@@ -328,10 +328,10 @@
       operational and telemetry information in the packet while the packet
       traverses a path between two points in the network. This document
       discusses the data fields and associated data types for in-situ OAM.
-      In-situ OAM data fields can be embedded into a variety of transports
-      such as NSH, Segment Routing, Geneve, native IPv6 (via extension
-      header), or IPv4. In-situ OAM can be used to complement OAM mechanisms
-      based on e.g. ICMP or other types of probe packets.</t>
+      In-situ OAM data fields can be embedded into a variety of protocols such
```
Comment (brockners) [inline drafts/draft-ietf-ippm-ioam-data.xml]: s/embedded/encapsulated/

We use "encapsulated" at the term for embedding IOAM into protocols throughout the IOAM specs

**Comment by brockners** [comment]
LGTM

**Pull request merged**